lib/pdk/util.rb

```diff
@@ -196,50 +196,154 @@ def targets_relative_to_pwd(targets)
     end
     module_function :targets_relative_to_pwd
 
-    def default_template_url
-      answer_file_url = PDK.answers['template-url']
-
-      return puppetlabs_template_url if answer_file_url.nil?
+    # @returns String
```

Comment:
I'm not a huge fan of throwing all these related methods into PDK::Util. I think it would be cleaner to make most of these instance methods of a new class `PDK::Util::TemplateURI` that takes a uri on initialisation, and then `template_uri` and `templates` could be class methods on `PDK::Util::TemplateURI` that operate on instances of the class.

eg `PDK::Util::TemplateURI.first_valid_template(options)` (renaming `template_uri` here) returns an instance of `PDK::Util::TemplateURI`.

Reply:
Good idea! I started with just one method in `PDK::Util` then it grew and grew. Being able to pass a TemplateURI instance around would be nice.